ingestion-beam/src/main/java/com/mozilla/telemetry/util/Normalize.java

```diff
@@ -4,28 +4,54 @@
 
 package com.mozilla.telemetry.util;
 
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
 
 public class Normalize {
 
+  private static final String OTHER = "Other";
+
   private static final String BETA = "beta";
   private static final String NIGHTLY = "nightly";
-  private static final String OTHER = "Other";
   private static final List<String> CHANNELS = ImmutableList.of("release", "esr", BETA, "aurora",
       NIGHTLY);
 
+  private static final String WINDOWS = "Windows";
+  private static final String MAC = "Mac";
+  private static final String LINUX = "Linux";
+  private static final String IOS = "iOS";
+  private static final String ANDROID = "Android";
+
+  private static final List<String> APP_NAMES = ImmutableList.of(
+      // Make sure more-specific versions come before Zerda.
+      "Zerda_cn", "Zerda",
+      // Maker suer more-specific versions come before Focus.
+      "Focus-TV", "Focus", "Klar",
+      // FirefoxRealityX
+      "FirefoxReality_oculusvr", "FirefoxReality_googlevr", "FirefoxReality_wavevr",
+      // FirefoxX
+      "FirefoxForFireTV", "FirefoxConnect",
+      // Other apps
+      "Fennec", "Scryer", "WebXR", "Lockbox",
+      // Desktop Firefox
+      "Firefox");
+
+  private static final Pattern OS_VERSION_RE = Pattern.compile("([0-9]+(?:\\.[0-9]+){0,2}).*");
+
   /**
    * Normalize a channel string to one of the five canonical release channels or "Other".
    *
-   * <p>Reimplementation of https://github.com/mozilla-services/lua_sandbox_extensions/blob/sprint2018-6/moz_telemetry/modules/moz_telemetry/normalize.lua#L167-L178
+   * <p>Reimplementation of https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L167-L178
    */
   public static String channel(String name) {
+    name = Strings.nullToEmpty(name);
     if (CHANNELS.contains(name)) {
       // We could lowercase before doing this check, but don't to maintain historical practice.
       return name;
-    } else if (name == null) {
-      return OTHER;
```

Comment:
Changes to the `channel` method here are to better match the style used in the new methods introduced in this PR.